elements/rh-avatar/rh-avatar.ts

```diff
@@ -1,32 +1,303 @@
-import { LitElement, html } from 'lit';
-import { customElement } from 'lit/decorators.js';
+import { customElement, property } from 'lit/decorators.js';
 
-import { pfelement } from '@patternfly/pfe-core/decorators.js';
+import { BaseAvatar } from '@patternfly/pfe-avatar/BaseAvatar.js';
+
+import { hash } from './lib/djb-hash';
+import { hsl2rgb, rgb2hsl } from './lib/hslrgb';
 
 import styles from './rh-avatar.css';
 
+import { colorContextProvider } from '../../lib/context/color.js';
+
+type Vector2D = [x: number, y: number];
+type Colors = Record<`color${number}`, string>;
+
+/** Ensure avatar colours are registered before custom elements upgrade */
+function register(klass: typeof RhAvatar) {
+  klass.registerColors();
+}
+
 /**
  * An avatar is a visual used to represent a user. It may contain an image or
  * a placeholder graphic.
  *
  *
  * @summary  An avatar is a visual used to represent a user.
  */
-@customElement('rh-avatar') @pfelement()
-export class RhAvatar extends LitElement {
+@customElement('rh-avatar')
+export class RhAvatar extends BaseAvatar {
   static readonly version = '{{version}}';
 
   static readonly styles = [styles];
 
+  private static readonly defaultSize = 128;
+
+  private static readonly defaultColors = '#67accf #448087 #709c6b #a35252 #826cbb';
```

Comment:
They don't match our standard colors and don't seem to be used anywhere else. Do you think we should add tokens for them?

Reply:
Summarizing Benny's response in chat here to keep track of everything together: It would be best for the avatar colors to be existing crayon tokens. To accomplish this, we could either change the avatar colors, or add new crayon tokens.